### PR TITLE
Eltek enexus

### PR DESCRIPTION
**Pull request by barryodonovan**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

### CODE REVIEW

File: includes/definitions/eltek.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+type: power
+icon: eltek
+#sysObjectID: .1.3.6.1.4.1.12148
+nobulk: 1
+mib_dir:
+    - eltek
+over:
+    - { graph: device_bits, text: 'Device Traffic' }
+discovery:
+    - .1.3.6.1.4.1.12148.10.2.4.0
```
Comment (murrant) [inline includes/definitions/eltek.yaml]: This isn't valid.  remove the discovery section.

Comment (barryodonovan) [reply includes/definitions/eltek.yaml]: Gone, but now I don't understand how it is doing discovery!

Comment (murrant) [reply includes/definitions/eltek.yaml]: check #7558

**Comment by murrant** [comment]
You might be able to use #7558 if/when it is merged.

File: includes/polling/os/eltek.inc.php [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * @package    LibreNMS
+ * @link       http://librenms.org
+ * @copyright  2017 Tony Murray
+ * @author     Tony Murray <murraytony@gmail.com>
+ */
+
+$hardware = snmp_get($device, 'powerSystem.powerSystemModel.0', '-Ovqa', 'SP2-MIB') . ' ' . snmp_get($device, 'powerSystem.powerSystemNominalVoltage.0', '-Ovqa', 'SP2-MIB'). 'v Nominal voltage with ' . snmp_get($device, 'rectifiers.rectifiersNumberOfRectifiers.0', '-Ovqa', 'SP2-MIB') . ' rectifiers of type ' . snmp_get($device, 'rectifiers.rectifierTable.rectifierEntry.rectifierType.1', '-Ovqa', 'SP2-MIB');
```
Comment (laf) [inline includes/polling/os/eltek.inc.php]: Please convert all snmp_gets into a single snmp_get_multi_oid(). However I'm not sure that hardware value is right. This is used to distinguish the hardware at a basic level (used in searches and the dropdown on device pages) so something this long is ott imho.

File: includes/polling/os/eltek.inc.php [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+$version = snmp_get($device, 'controlSystem.controlSystemInventory.controlUnitTable.controlUnitEntry.controlUnitSwVersion.1', '-Ovqa', 'SP2-MIB');
+//$features = implode(', ', explode(PHP_EOL, snmp_walk($device, 'vpwrModuleOID', '-Oqvs', 'EltekEnexusPowersystem_branch10_V37')));
```
Comment (laf) [inline includes/polling/os/eltek.inc.php]: Delete this if it's of no use.

File: tests/snmpsim/eltek.snmprec [hunk 4]
```diff
@@ -0,0 +1 @@
+.1.3.6.1.4.1.12148.10.2.4.0|4|Eltek
```
Comment (laf) [inline tests/snmpsim/eltek.snmprec]: Is this the one that doesn't have a sysObjectId or sysDescr?

Comment (barryodonovan) [reply tests/snmpsim/eltek.snmprec]: Yes indeed.

**Comment by barryodonovan** [comment]
@laf @murrant 
I'm quite happy to change it, it only covers a subset of the Eltek-Valere family.
Minicom is the model I have, eNexus is the product family, SP2 is the MIB, I'm not sure which is better? There is no obvious firmware name.

Perhaps, Eltek-eNexus? That's what I've called the branch for the PR as I think it is most accurate?

**Comment by laf** [comment]
I'd go for just `enexus`

**Comment by barryodonovan** [comment]
I'll push it tomorrow morning
________________________________

**Comment by barryodonovan** [comment]
Back to failing on the pre-commit script again.

Can't see why, last time it was the cache invalidation issue but not now.

**Comment by laf** [comment]
Most likely because we had to remove the snmp_get ability in discovery, it's back in again now so just rebase and it should pass.

**Comment by laf** [comment]
:( This has ended up with loads of commits we've already merged :(

**Comment by barryodonovan** [comment]
Hopefully this rebase is a bit cleaner!

**Pull request merged**